src/Microsoft.AspNetCore.Server.Kestrel/BadHttpRequestException.cs

```diff
@@ -102,6 +102,9 @@ internal static BadHttpRequestException GetException(RequestRejectionReason reas
                 case RequestRejectionReason.TooManyHeaders:
                     ex = new BadHttpRequestException("Request contains too many headers.", 431);
                     break;
+                case RequestRejectionReason.HttpsOnHttp:
+                    ex = new BadHttpRequestException("An SSL/TLS handshake might have been attempted at an HTTP endpoint.", 400);
```

Comment:
I think this should read `An SSL/TLS handshake might have been attempted **on** an HTTP endpoint`